docs/sources/operators-guide/configuring/reference-configuration-parameters/index.md

```diff
@@ -1400,6 +1400,10 @@ query_frontend:
   # CLI flag: -ruler.query-frontend.address
   [address: <string> | default = ""]
 
+  # The timeout for a rule query being evaluated by the query-frontend.
+  # CLI flag: -ruler.query-frontend.timeout
+  [timeout: <duration> | default = 2m]
```

Comment:
I'm not convinced about having yet another timeout config option. Why not reusing `-querier.timeout`, documenting that needs to be set for ruler too when remote rule evaluation is used? We already use it in query-frontend too.

Reply:
Created an issue: https://github.com/grafana/mimir/issues/2129

Reply:
`-querier.timeout` is poorly named (perhaps "-query.timeout" would be better) but it would make sense to reuse it, since ruler already uses it when it's not configured to use query-frontend.

Reply:
I agree it's poorly named, but unfortunately changing it would be a breaking change (so to rename we would have to go through a deprecation process keeping backward compatibility for 2 releases). For the sake of config reduction, I think just reusing it is enough.